management-api-agent-common/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java

```diff
@@ -739,96 +742,101 @@ public String repair(
       @RpcParam(name = "keyspaceName") String keyspace,
       @RpcParam(name = "tables") List<String> tables,
       @RpcParam(name = "full") Boolean full,
-      @RpcParam(name = "notifications") boolean notifications)
+      @RpcParam(name = "notifications") boolean notifications,
+      @RpcParam(name = "repairParallelism") Optional<RepairParallelism> repairParallelism,
+      @RpcParam(name = "datacenters") Optional<Collection<String>> datacenters,
+      @RpcParam(name = "associatedTokens") Optional<List<RingRange>> associatedTokens,
+      @RpcParam(name = "repairThreadCount") Optional<Integer> repairThreadCount)
       throws IOException {
     // At least one keyspace is required
-    if (keyspace != null) {
-      // create the repair spec
-      Map<String, String> repairSpec = new HashMap<>();
-
-      // add any specified tables to the repair spec
-      if (tables != null && !tables.isEmpty()) {
-        // set the tables/column families
-        repairSpec.put(RepairOption.COLUMNFAMILIES_KEY, String.join(",", tables));
-      }
-
-      // handle incremental vs full
-      boolean isIncremental = Boolean.FALSE.equals(full);
-      repairSpec.put(RepairOption.INCREMENTAL_KEY, Boolean.toString(isIncremental));
-      if (isIncremental) {
-        // incremental repairs will fail if parallelism is not set
-        repairSpec.put(RepairOption.PARALLELISM_KEY, RepairParallelism.PARALLEL.getName());
-      }
+    assert (keyspace != null);
+    Map<String, String> options = new HashMap<>();
+    repairParallelism.map(rPar -> options.put(RepairOption.PARALLELISM_KEY, rPar.getName()));
+    options.put(RepairOption.INCREMENTAL_KEY, Boolean.toString(!full));
+    repairThreadCount.map(
+        tCount ->
+            options.put(RepairOption.JOB_THREADS_KEY, Integer.toString(tCount == 0 ? 1 : tCount)));
+    options.put(RepairOption.TRACE_KEY, Boolean.toString(Boolean.FALSE));
+    options.put(RepairOption.COLUMNFAMILIES_KEY, StringUtils.join(tables, ","));
```

Comment:
This is a good question and something I was discussing with the team last night. I think if the argument is optional I'd rather wrap it in an Optional<> in this part of the codebase. What do you think? I didn't previously do this to avoid changing the existing handling, but as I've already done that, it seems like maybe I should just bite the bullet and use Optional<>...

Then yes, you have the other two cases where the List is empty or the argument itself is null. Dealing with all of these cases is honestly quite a headache as I'm not sure how things get deserialised from json!

Reply:
I think we've discussed this. We will remove the Optionals<> depending on Olivier's feedback. For now I'll keep them and expand the input sanitation.